Loop/Extensions/UserDefaults+Loop.swift

```diff
@@ -130,4 +131,23 @@ extension UserDefaults {
             }
         }
     }
+    
+    var automaticDosingStatus: AutomaticDosingStatus? {
```

Comment:
AutomaticDosingStatus should be computable at any time from automaticDosingEnabled and isAutomaticDosingAllowed, so I'm not sure I understand why additional persistence is needed. But we are going to probably be reviewing this again with the automation design updates.

Reply:
Let me see why this was needed 👀

Reply:
I'll dig a bit deeper into this when I'm back from OOO but for now I'll merge as-is!